File: 132P/primes.java
public class primes {
  static int SIZE = 500;
  static int LINEZ = 50;
  static int[] prime;
  static boolean isPrime(int n)
  {
    for (int k = 1;; ++k) {
      int p = prime[k];
      if (n%p == 0) return false;
      if (p >= n/p) return true;
    }
  }
  public static void main(String[] args)
  {
    System.out.println("Table of first 500 numbers");
    prime = new int[SIZE];
    prime[0] = 2;
    prime[1] = 3;
    for (int n = 5, j = 2; j < SIZE; n += 2) {
      if (isPrime(n)) prime[j++] = n;
    }
    for (int i = 0; i < LINEZ; ++i) {
      String s = "";
      for (int j = i; j < SIZE; j += LINEZ) {
        s += String.format(" %04d", prime[j]);
      }
      System.out.println(s);
    }
  }
}
// log:

File: 132P/primes.js
#!/usr/bin/env node
function spit(prime, linez) {
  console.log("Table of first 500 primes:");
  for (let i = 0; i < linez; ++i) {
    let s = "";
    for (let j = i; j < prime.length; j += linez) {
      s += " " + ("000" + prime[j]).slice(-4); 
    }
    console.log(s);
  }
}
/* n - next prime candidate *
 * p - list of found primes */
function isPrime(n, p) {
  for (let k = 1;; ++k) {
    if (n%p[k] == 0   ) return false;
    if (n/p[k] <= p[k]) return true ; // thats
  }
}
function primes() {
  let prime = [2,3];
  for (let n = 5; prime.length < 500; n += 2) {
    if (isPrime(n, prime)) {
      prime.push(n);
    }
  }
  spit(prime, 50);
}
primes();
